lib/api_me/sorting.rb

```diff
@@ -26,20 +31,22 @@ def sort_meta
 
     protected
 
-    def sort
-      unless sort_criteria === ""
-        sort_p = sort_criteria
+    def sort(criteria = default_sort_criteria)
+      unless sort_association == ""
+        criteria_class = criteria.camelize.constantize
+
         if sort_reverse === "true"
-          self.scope = scope.sort_by {|scope| scope[sort_p]}.reverse!
+          scope.joins(criteria.to_sym).merge(criteria_class.order(sort_association => :desc))
         else
-          self.scope = scope.sort_by {|scope| scope[sort_p]}
+          scope.joins(criteria.to_sym).merge(criteria_class.order(sort_association => :asc))
```

Comment:
Sorry, `left_outer_joins` is only available in 5+, not 4.2.  You can use squeels outer join method (https://github.com/activerecord-hackery/squeel#joins) or manually do `.joins("LEFT OUTER JOIN relation_table_name.id ON fk_field_name")`